packages/game-passes/contracts/GamePasses.sol

```diff
@@ -0,0 +1,1161 @@
+            }
+        }
+    }
+
+    /**
+     * @notice Helper function to check if minting would exceed max per wallet
+     * @param tokenId The token ID to check
+     * @param to The recipient address
+     * @param amount The amount to mint
+     * @dev Used internally before user mint operations
+     * @dev Reverts if:
+     *      - Current wallet balance + amount would exceed max per wallet
+     */
+    function _checkMaxPerWallet(uint256 tokenId, address to, uint256 amount) private view {
+        TokenConfig storage config = tokenConfigs[tokenId];
+        if (config.mintedPerWallet[to] + amount > config.maxPerWallet) {
+            revert ExceedsMaxPerWallet(tokenId, to, amount, config.maxPerWallet);
+        }
+    }
+
+    // =============================================================
+    //                   Royalties (ERC2981)
+    // =============================================================
+
+    /**
+     * @notice Sets the default royalty info (applies to all token IDs)
+     * @param receiver Address that will receive the royalties
+     * @param feeNumerator Royalty amount in basis points (1% = 100)
+     * @dev Only callable by addresses with ADMIN_ROLE
+     * @dev Updates the default royalty that applies to all tokens
+     * @dev Can be overridden per token using setTokenRoyalty
+     * @dev Reverts if:
+     *      - Caller doesn't have ADMIN_ROLE
+     */
+    function setDefaultRoyalty(address receiver, uint96 feeNumerator) external onlyRole(ADMIN_ROLE) {
+        _setDefaultRoyalty(receiver, feeNumerator);
+    }
+
+    /**
+     * @notice Sets royalty info for a specific token ID, overriding default royalty
+     * @param tokenId ID of the token to set royalties for
+     * @param receiver Address that will receive the royalties for this token
+     * @param feeNumerator Royalty amount in basis points (1% = 100)
+     * @dev Only callable by addresses with ADMIN_ROLE
+     * @dev Overrides the default royalty for the specified token ID
+     * @dev Reverts if:
+     *      - Caller doesn't have ADMIN_ROLE
+     */
+    function setTokenRoyalty(uint256 tokenId, address receiver, uint96 feeNumerator) external onlyRole(ADMIN_ROLE) {
+        _setTokenRoyalty(tokenId, receiver, feeNumerator);
+    }
+
+    // =============================================================
+    //                   ERC1155 Overrides
+    // =============================================================
+
+    /**
+     * @notice Returns the metadata URI for a specific token ID
+     * @param tokenId ID of the token to get URI for
+     * @dev Constructs the URI by concatenating baseURI + tokenId + ".json"
+     * @dev Can be overridden by derived contracts to implement different URI logic
+     * @return string The complete URI for the token metadata
+     */
+    function uri(uint256 tokenId) public view virtual override returns (string memory) {
+        return string(abi.encodePacked(baseURI, tokenId.toString(), ".json"));
```

Comment:
this will return a string for all tokenIds, however we need to have some sort of check that tokenId exists - eg for tokenId 0, or other ids that haven't been created

Reply:
add `function exists(id)` ?

Reply:
Such function is already implemented through `ERC1155SupplyUpgradeable` extension

However I feel like we should override it to check if the token has been configured.
I have implemented it here: https://github.com/thesandboxgame/sandbox-smart-contracts/pull/1619/commits/657e0258a34afc437f0a4f44223fc873f16022db